codersdk/workspaces.go

```diff
@@ -43,6 +43,30 @@ type Workspace struct {
 	// unlocked by an admin. It is subject to deletion if it breaches
 	// the duration of the locked_ttl field on its template.
 	LockedAt *time.Time `json:"locked_at" format:"date-time"`
+	// Health reports the health of the workspace and its agents.
+	Health WorkspaceHealth `json:"health"`
+}
+
+type WorkspaceHealth struct {
+	Healthy         bool                               `json:"healthy"`          // Healthy is true if the workspace and all of its agents are healthy.
```

Comment:
This will work until we've only `healthy`/`unhealthy`. What if we decide to add more states, for instance:`unknown`, `recovering`, `zombie`, `tainted`, etc.

Maybe we should replace the `healthy` with `state: healthy`?

Reply:
Based on https://github.com/coder/coder/issues/6461#issuecomment-1612826927, we don't want more values in the enum. All of the mentioned states (`unknown`, `recovering`, ...) would work under `unhealthy` with the reason field filled out (or similar).